File: LogisimPlus/LogisimPlus/logisim/gui/generic/LFrame.cs
// ====================================================================================================
// Produced by the Free Edition of Java to C# Converter.
// To produce customized conversions, purchase a Premium Edition license:
// https://www.tangiblesoftwaresolutions.com/product-details/java-to-csharp-converter.html
// ====================================================================================================

using System;
using System.Collections.Generic;

/* Copyright (c) 2010, Carl Burch. License information is located in the
 * logisim.Main source code and at www.cburch.com/logisim/. */

namespace logisim.gui.generic
{


	using WindowClosable = logisim.util.WindowClosable;

	public class LFrame : JFrame, WindowClosable
	{
		private const string PATH = "logisim/img/logisim-icon-";
		private static readonly int[] SIZES = new int[] {16, 20, 24, 48, 64, 128};
		private static IList<Image> ICONS = null;
		private const int DEFAULT_SIZE = 48;
		private static Image DEFAULT_ICON = null;

		public static void attachIcon(Window frame)
		{
			if (ICONS == null)
			{
				IList<Image> loadedIcons = new List<Image>();
				ClassLoader loader = typeof(LFrame).getClassLoader();
				foreach (int size in SIZES)
				{
					URL url = loader.getResource(PATH + size + ".png");
					if (url != null)
					{
						ImageIcon icon = new ImageIcon(url);
						loadedIcons.Add(icon.getImage());
						if (size == DEFAULT_SIZE)
						{
							DEFAULT_ICON = icon.getImage();
						}
					}
				}
				ICONS = loadedIcons;
			}

			bool success = false;
			try
			{
				if (ICONS != null && ICONS.Count > 0)
				{
					System.Reflection.MethodInfo set = frame.GetType().GetMethod("setIconImages", typeof(System.Collections.IList));
					set.invoke(frame, ICONS);
					success = true;
				}
			}
			catch (Exception)
			{
			}

			if (!success && frame is JFrame && DEFAULT_ICON != null)
			{
				((JFrame) frame).setIconImage(DEFAULT_ICON);
			}
		}

		public LFrame()
		{
			LFrame.attachIcon(this);
		}

		public virtual void requestClose()
		{
			WindowEvent closing = new WindowEvent(this, WindowEvent.WINDOW_CLOSING);
			processWindowEvent(closing);
		}
	}

}

File: logisim/app/src/main/java/logisim/gui/generic/LFrame.java
/* Copyright (c) 2010, Carl Burch. License information is located in the
 * logisim.Main source code and at www.cburch.com/logisim/. */

package logisim.gui.generic;

import java.awt.Image;
import java.awt.Window;
import java.awt.event.WindowEvent;
import java.lang.reflect.Method;
import java.net.URL;
import java.util.ArrayList;
import java.util.List;

import javax.swing.ImageIcon;
import javax.swing.JFrame;

import logisim.util.WindowClosable;

public class LFrame extends JFrame implements WindowClosable {
	private static final String PATH = "logisim/img/logisim-icon-";
	private static final int[] SIZES = { 16, 20, 24, 48, 64, 128 };
	private static List<Image> ICONS;
	private static final int DEFAULT_SIZE = 48;
	private static Image DEFAULT_ICON;

	public static void attachIcon(Window frame) {
		if (ICONS == null) {
			List<Image> loadedIcons = new ArrayList<>();
			ClassLoader loader = LFrame.class.getClassLoader();
			for (int size : SIZES) {
				URL url = loader.getResource(PATH + size + ".png");
				if (url != null) {
					ImageIcon icon = new ImageIcon(url);
					loadedIcons.add(icon.getImage());
					if (size == DEFAULT_SIZE) DEFAULT_ICON = icon.getImage();
				}
			}
			ICONS = loadedIcons;
		}

		boolean success = false;
		try {
			if (!ICONS.isEmpty()) {
				Method set = frame.getClass().getMethod("setIconImages", List.class);
				set.invoke(frame, ICONS);
				success = true;
			}
		}
		catch (Exception e) {
		}

		if (!success && frame instanceof JFrame && DEFAULT_ICON != null) frame.setIconImage(DEFAULT_ICON);
	}

	public LFrame() {
		LFrame.attachIcon(this);
	}

	public void requestClose() {
		WindowEvent closing = new WindowEvent(this, WindowEvent.WINDOW_CLOSING);
		processWindowEvent(closing);
	}
}
